vis-lua.c

```diff
@@ -832,6 +832,9 @@ static int keymap(lua_State *L, Vis *vis, Win *win) {
 			goto err;
 	} else if (lua_isuserdata(L, 4)) {
 		binding->action = obj_ref_check(L, 4, VIS_LUA_TYPE_KEYACTION);
+	} else {
+		vis_binding_free(vis, binding);
+		luaL_argerror(L, 4, "invalid handler");
```

Comment:
I think this should just be `goto err;`. If a user ends up confused
as to why their mapping didn't work they can consult the documentation
and see that `vis:map()` and `win:map()` return a boolean indicating
success. We also have `vis:unmap()` for removing a binding so if
that was the goal it should just be called instead.